routers/repo/editor_test.go

```diff
@@ -15,14 +15,15 @@ func TestCleanUploadName(t *testing.T) {
 	models.PrepareTestEnv(t)
 
 	var kases = map[string]string{
-		".git/refs/master": "git/refs/master",
+		".git/refs/master": "",
 		"/root/abc":        "root/abc",
 		"./../../abc":      "abc",
-		"a/../.git":        "a/.git",
-		"a/../../../abc":   "a/abc",
+		"a/../.git":        "",
+		"a/../../../abc":   "abc",
 		"../../../acd":     "acd",
-		"../../.git/abc":   "git/abc",
-		"..\\..\\.git/abc": "git/abc",
+		"../../.git/abc":   "",
+		"..\\..\\.git/abc": "..\\..\\.git/abc",
```

Comment:
So because we never actually write the files this is completely safe and it accords with what git would do.

It's perfectly valid to have a git file with backslashes in it.

Reply:
How will this work on Windows? :/

Reply:
Absolutely fine. The files are never checked out. I add directly to the index and never let a user defined filename touch the filesystem. 

I'm not sure what git does when it tries to checkout an evil file like that on Windows, but looking at the source code for index there's no special protection applied to anything except the .git file.

This is the point of moving to this index approach - people can be as pathological in their repos as they like and we won't get bitten. At present if you try to merge a repo with one of these pathological files in it you'll get bitten on Windows. You just need to create it on the command line and push it.

Reply:
can you add also test case for something like `..\\../.git/abc`?

Reply:
Added.